### PR TITLE
feat: (ecam) Add Hydraulics ECAM page

### PR DESCRIPTION
**Pull request by RichardPilbery**
Fixes #[issue_no]

## Summary of Changes

Compliment to existing PR (#3782) implementing the hydraulics system on the A32NX. This is the ECAM page for the hydraulics system, written in React. Note that this PR should not be merged before #3782 

## Screenshots (if necessary)

ECAM-HYD_ELEC-PB

ecam-hyd-v4

## References

HYD-page-reference

ECAM-HYD-INFLIGHT

## Additional context

Discord username (if different from GitHub): tricky_dicky[z+0]

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by wpine215** [comment]
Only thing that stands out to me at the moment is that the "HYD" title seems to be positioned a bit lower than it is IRL

**Comment by RichardPilbery** [comment]
> Only thing that stands out to me at the moment is that the "HYD" title seems to be positioned a bit lower than it is IRL

Yeah, but if it is moved up any higher, you lose the top of the letters. It's on my things to investigate list.

File: src/instruments/src/SD/Pages/Hyd/Hyd.tsx [hunk 1]
```diff
@@ -25,25 +25,42 @@ import { SimVarProvider, useSimVar } from '../../../Common/simVars';
...
-    const [yellowFireValve] = useSimVar('L:A32NX_HYD_YELLOW_FIRE_VALVE_OPENED', 'boolean', 1000);
+    const [greenFireValve] = useSimVar('L:A32NX_HYD_GREEN_FIRE_VALVE_OPENED', 'boolean', 500);
+    const [yellowFireValve] = useSimVar('L:A32NX_HYD_YELLOW_FIRE_VALVE_OPENED', 'boolean', 500);
+
+    const [greenPumpActive] = useSimVar('L:A32NX_HYD_GREEN_EDPUMP_ACTIVE', 'boolean', 500);
+    const [yellowPumpActive] = useSimVar('L:A32NX_HYD_YELLOW_EDPUMP_ACTIVE', 'boolean', 500);
+    const [bluePumpActive] = useSimVar('L:A32NX_HYD_BLUE_EPUMP_ACTIVE', 'boolean', 500);
+
+    const [greenPumpLowPressure] = useSimVar('L:A32NX_HYD_GREEN_EDPUMP_LOW_PRESS', 'boolean', 500);
+    const [yellowPumpLowPressure] = useSimVar('L:A32NX_HYD_GREEN_EDPUMP_LOW_PRESS', 'boolean', 500);
```
Comment (crocket63) [inline src/instruments/src/SD/Pages/Hyd/Hyd.tsx]: Meant YELLOW_EDPUMP here? :)

Comment (RichardPilbery) [reply src/instruments/src/SD/Pages/Hyd/Hyd.tsx]: Affirm. Good spot.

Comment (RichardPilbery) [reply src/instruments/src/SD/Pages/Hyd/Hyd.tsx]: Actually, this has already been fixed. Odd that it is still showing for you...

Comment (crocket63) [reply src/instruments/src/SD/Pages/Hyd/Hyd.tsx]: Better this way than the other :D  Probably did a dumb click again :D

File: src/instruments/src/SD/Pages/Hyd/Hyd.tsx [hunk 2]
```diff
@@ -0,0 +1,767 @@
...
+
+    const [greenPressure] = useSimVar('L:A32NX_HYD_GREEN_PRESSURE', 'psi', 500);
+    const [yellowPressure] = useSimVar('L:A32NX_HYD_YELLOW_PRESSURE', 'psi', 500);
+    const [bluePressure] = useSimVar('L:A32NX_HYD_BLUE_PRESSURE', 'psi', 500);
+
+    const [greenPumpPBStatus] = useSimVar('L:A32NX_OVHD_HYD_ENG_1_PUMP_PB_IS_AUTO', 'boolean', 500);
+    const [yellowPumpPBStatus] = useSimVar('L:A32NX_OVHD_HYD_ENG_2_PUMP_PB_IS_AUTO', 'boolean', 500);
+    const [bluePumpPBStatus] = useSimVar('L:A32NX_OVHD_HYD_EPUMPB_PB_IS_AUTO', 'boolean', 500);
+
+    const [yellowElectricPumpStatus] = useSimVar('L:A32NX_OVHD_HYD_EPUMPY_PB_IS_AUTO', 'boolean', 500);
```
Comment (crocket63) [inline src/instruments/src/SD/Pages/Hyd/Hyd.tsx:40]: My feeling is that pump status should directly come from the XPUMP_ACTIVE simvars. For example if any logic starts a pump without changing button status, we won't see it active.  If yellow epump is activated by cargo door operation from ground crew, it won't appear on ecam currently. Is it expected?

Comment (RichardPilbery) [reply src/instruments/src/SD/Pages/Hyd/Hyd.tsx:40]: For the Yellow electrical pump you are right. However, the status of the push button is important too.

Comment (crocket63) [reply src/instruments/src/SD/Pages/Hyd/Hyd.tsx]: Totally agree, but my point is that I already evaluate the button and compute the pump activation logic inside the WASM system. So I think there is no point doing it the same at different places UNLESS the ecam logic is different from the actual pump logic. 
Probably need to ask for this on discord :)
As an example, you should also check for blue pump override to have it showing on the ECAM when pressed, and this is already done and handled in the WASM module.  Or at least if it's done wrong e only have to update it in one place :)

**Review by wpine215 — APPROVED** [review]

**Comment by wpine215** [comment]
Needs testing instructions then will be ready for testing.

**Comment by crocket63** [comment]
> Needs testing instructions then will be ready for testing.

Is it actually testable without hydraulics PR?  In a way it should, as issues could come from hydraulics and not the ecam stuff, but not sure how to do so?

**Comment by RichardPilbery** [comment]
> > Needs testing instructions then will be ready for testing.
> 
> Is it actually testable without hydraulics PR? In a way it should, as issues could come from hydraulics and not the ecam stuff, but not sure how to do so?

I don't think it is. Let's get the hydraulics PR merged and then we can test this.

File: src/instruments/src/SD/Pages/Hyd/Hyd.tsx [hunk 3]
```diff
@@ -0,0 +1,788 @@
...
+                setPressures(true);
+                setPtuActive(0);
+            }
+        } else if (ptuAvailable && yellowElectricPumpStatus && greenPressure <= 2990) {
+            setPtuScenario('right-to-left');
+            setPtuActive(1);
+        } else {
+            setPtuScenario(ptuAvailable ? 'normal' : 'PTU-off');
+        }
+    }, [greenPressure, yellowPressure, yellowElectricPumpStatus]);
```
Comment (davidwalschots) [inline src/instruments/src/SD/Pages/Hyd/Hyd.tsx:162]: **thought**: I hope @crocket63 will provide enough data to not require these calculation in the presentation layer. We should strive to keep the React side as lean as possible.

Comment (crocket63) [reply src/instruments/src/SD/Pages/Hyd/Hyd.tsx:162]: There's actually specific ecam logic that is not aware of the PTU state. If we go full simulation that logic might be ok to be here.

**Comment by crocket63** [comment]
I noticed one thing when testing: when first opening HYD page, you can see the page going from init state to actual state. Can it be kind of "refreshed" before it's actually shown?

**Comment by wpine215** [comment]
#3782 has now been merged.

File: src/instruments/src/SD/Pages/Hyd/Hyd.tsx (hunk 3 above)
Comment (crocket63) [reply src/instruments/src/SD/Pages/Hyd/Hyd.tsx:162]: If there's a true performance issue, it wouldn't be too difficult to process the exact same logic runing here on rust side just outputting the required ecam boolean A32NX_HYD_ECAM_GREEN_AVAILABLE  A32NX_HYD_ECAM_PTU_G_2_Y_ACTIVE etc....

**Pull request closed without merging**